src/Microsoft.AspNetCore.Razor.Evolution/CodeGeneration/CSharpRenderingContext.cs

```diff
@@ -60,6 +62,19 @@ public void AddLineMappingFor(RazorIRNode node)
                 return;
             }
 
+            var imports = CodeDocument.GetImportSyntaxTrees();
+            if (imports != null)
+            {
+                for (var i = 0; i < imports.Count; i++)
+                {
+                    if (string.Equals(imports[i].Source.FileName, node.Source.Value.FilePath, StringComparison.OrdinalIgnoreCase))
```

Comment:
I dunno, we haven't 'cleaned' this code yet, so I guess either one you want is fine.